pkg/authn/multikeychain.go

```diff
@@ -28,14 +28,50 @@ func NewMultiKeychain(kcs ...Keychain) Keychain {
 
 // Resolve implements Keychain.
 func (mk *multiKeychain) Resolve(target Resource) (Authenticator, error) {
+	var auths []Authenticator
 	for _, kc := range mk.keychains {
 		auth, err := kc.Resolve(target)
```

Comment:
I think here should call `ResolveMany`, Resolve only return the first secret matched, so if resolve only return one matched, use `Authenticator`